File: JavaScript solutions/11. Container With Most Water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function (height) {
  var mx = 0;
  var len = height.length;
  for (var i = 0, j = len - 1; i < j; ) {
    var ans = Math.min(height[i], height[j]);
    ans *= Math.abs(i - j);
    mx = Math.max(mx, ans);
    if (height[i] <= height[j]) i++;
    else j--;
  }
  return mx;
};

File: C++ solutions/11. Container With Most Water.cpp
#include<bits/stdc++.h>
using namespace std;

// solution 
class Solution {
public:
    int maxArea(vector<int>& height) {
        long long mx=0;
        int len=height.size();
        for(int i=0,j=len-1;i<j;){
            long long ans=min(height[i],height[j]);
            ans*=abs(i-j);
            mx=max(mx,ans);

            if(height[i]<=height[j]) i++;
            else j--;
        }
        return mx;
    }
};

